pkg/hierarchy/manager.go

```diff
@@ -0,0 +1,80 @@
+package hierarchy
+
+// Manager stores Cohorts and ClusterQueues, and maintains the edges
+// between them.
+type Manager[CQ cqNode[CQ, C], C cohortNode[CQ, C]] struct {
```

Comment:
Are the generic types to simplify the implementation, or they are in anticipation of future needs? 
If in anticipation of future needs I would suggest to defer them for later so that the PR just two fold responsibility : encapsulating the edge management, and deduplicate the code for edge management.

Reply:
They are needed because use this in both `cache` and `queue` packages.

Reply:
sgtm